test/ml_dsa_test.c

```diff
@@ -194,6 +199,286 @@ static int ml_dsa_sigver_test(int tst_id)
     return ret;
 }
 
+static int ml_dsa_key_dup_test(void)
+{
+    int ret = 0;
+    const ML_DSA_KEYGEN_TEST_DATA *tst = &ml_dsa_keygen_testdata[0];
+    EVP_PKEY *pkey = NULL, *pkey_copy = NULL;
+    EVP_PKEY_CTX *ctx = NULL;
+
+    if (!TEST_ptr(pkey = do_gen_key(tst->name, tst->seed, tst->seed_len)))
+        goto err;
+    if (!TEST_ptr(pkey_copy = EVP_PKEY_dup(pkey)))
+        goto err;
+    if (!TEST_int_eq(EVP_PKEY_eq(pkey, pkey_copy), 1))
+        goto err;
+    if (!TEST_ptr(ctx = EVP_PKEY_CTX_new_from_pkey(lib_ctx, pkey_copy, NULL)))
+        goto err;
+    if (!TEST_int_eq(EVP_PKEY_check(ctx), 1))
+        goto err;
```

Comment:
We'd normally express this as a string of `||`s.
Likewise in the functions that follow.